File: src/main/scala/io/github/nwtgck/gh_card/domain/GitHubApiService.scala
package io.github.nwtgck.gh_card.domain

import io.github.nwtgck.gh_card.GitHubJson

import scala.util.Try

trait GitHubApiService {
  /**
    * Get a repository
    * @param repoName
    * @return
    */
  def getRepository(repoName: String): Try[GitHubJson.Repository]
}

File: backend/src/domain/GitHubApiService.ts
import {GithubRepoJson} from '../types';


export interface GitHubApiService {
  /**
   * Get GitHub repository JSON via GitHub API
   * @param repoName owner/repo
   */
  getRepository(repoName: string): Promise<{repo: GithubRepoJson} | {status: number, resText: string}>;
}
